### ISSUE TITLE
tap2rpm has an unused "unsigned" command-line option.

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4086]
|<img alt="TimAllen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| TimAllen reported|
|-|-|
|Trac ID|trac#4086|
|Type|enhancement|
|Created|2009-11-12 05:23:03Z|
|Branch|https://github.com/twisted/twisted/tree/deprecate-tap2rpm-4086|

`twisted.scripts.tap2rpm.MyOptions` defines a flag called "unsigned" that does not appear to be used anywhere. I believe RPMs are unsigned unless you pass --sign to rpmbuild, so a special flag seems unnecessary.

Attachments:

* [remove_unsigned.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/22d079c78762448807e6f9babee1e7efb18f77ad.patch) (396 bytes) - added by jtanis on 2010-02-21 22:58:28Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/beed96fe1ccd7cb9040f7efc63fe82b767a01d67.patch) (396 bytes) - added by jtanis on 2010-02-21 22:58:33Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.3.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/32ca6c4835e41a617546a673fcb4ad7577200c9d.patch) (396 bytes) - added by jtanis on 2010-02-21 22:58:39Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.4.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/219b17b094d24d9c1d5baf788d66ea987225fef6.patch) (396 bytes) - added by jtanis on 2010-02-21 22:58:44Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.5.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/e911dd476e8c8e0f0b8ad35259b440ae55fda320.patch) (396 bytes) - added by jtanis on 2010-02-21 22:58:47Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.6.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/2ddcaa65a0d9287bc8fea907f50db5d3b647c4c1.patch) (396 bytes) - added by jtanis on 2010-02-21 22:58:48Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.7.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/fe94dc7541c9666315a44789e7ce71d2e5a0ce3c.patch) (396 bytes) - added by jtanis on 2010-02-21 22:58:56Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.8.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/11aae2b7e52893ec4316cbc21d65fa53bc229c8e.patch) (396 bytes) - added by jtanis on 2010-02-21 22:58:58Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.9.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/8657069d0b36b301cb48b2acaff948eceaf652e4.patch) (396 bytes) - added by jtanis on 2010-02-21 22:59:08Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.10.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/42951109fc57fc2351f41ef0ef428535fb1c5145.patch) (396 bytes) - added by jtanis on 2010-02-21 22:59:13Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.11.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/45a7213de4affa2ae7263562220b6f0efa57e641.patch) (396 bytes) - added by jtanis on 2010-02-21 22:59:20Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.12.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/5b9a2f28937e9cff20b03cabaae6f32cbbed812e.patch) (396 bytes) - added by jtanis on 2010-02-21 22:59:21Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.13.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/04242ebd3a51df0f4e8726cae8267c03d895128e.patch) (396 bytes) - added by jtanis on 2010-02-21 22:59:22Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.14.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/52b06389cce21ed9bf54e1ad3f00ceb2fd5f612b.patch) (396 bytes) - added by jtanis on 2010-02-21 22:59:22Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.15.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/663f4cce03b09002bc53d25c0bae9fb78d13bf41.patch) (396 bytes) - added by jtanis on 2010-02-21 22:59:23Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.16.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/ee04734c3ad57e7bcf04fefdf5965bdfcef173e7.patch) (396 bytes) - added by jtanis on 2010-02-21 22:59:23Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.17.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/24a3a96325ada7edc92c8672a19a3de9906bee01.patch) (396 bytes) - added by jtanis on 2010-02-21 22:59:25Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.18.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/4515c8329255300b990369bbe8f13ba28893c6c0.patch) (396 bytes) - added by jtanis on 2010-02-21 22:59:26Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.19.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/790891b717af277d4198da76bb8f00ecd7319bec.patch) (396 bytes) - added by jtanis on 2010-02-21 22:59:27Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.20.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/304bc88a5e970a6c0c656b0c7159a667b9efbdc1.patch) (396 bytes) - added by jtanis on 2010-02-21 22:59:32Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.21.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/e147108efbe5b9574544ffa7b56ab3cf89b2e6bf.patch) (396 bytes) - added by jtanis on 2010-02-21 22:59:42Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.22.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/461614e82d2613e18a1d41892e0a111dd18bca84.patch) (396 bytes) - added by jtanis on 2010-02-21 22:59:42Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.23.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/1f0a82f09d289f0143e298de639bf53ba56b0297.patch) (396 bytes) - added by jtanis on 2010-02-21 22:59:43Z - removed optFlags altogether, since unsigned is the only option.
* [remove_unsigned.24.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/74b57df30acda0c16716b61fc308c3ff40c5937a.patch) (396 bytes) - added by jtanis on 2010-02-21 22:59:43Z - removed optFlags altogether, since unsigned is the only option.
* [twisted4086.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/50caf17f155aa869288e89692e09549072f40d72.patch) (2279 bytes) - added by necaris on 2012-03-13 06:30:22Z - Patch to deprecate "unsigned" flag, with test
* [twisted4086.2.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/ad791da266e0e3aa2f84dd6c6585f5bc1a03063f.patch) (2878 bytes) - added by necaris on 2012-03-13 17:03:37Z - Patch to deprecate "unsigned" flag, with test and news file
* [twisted4086.3.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/0816b93838ed995de503219057f92c73c50c2f8a.patch) (5230 bytes) - added by necaris on 2012-03-13 17:10:07Z - Patch to deprecate "unsigned" flag, with test and news file, maintaining short-form option
* [twisted4086.4.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/e2a07cc79e66265a38d51d8b202ea18025f1226b.patch) (5230 bytes) - added by necaris on 2012-03-13 17:25:17Z - Patch to deprecate "unsigned" flag, with test and news file, corrected deprecating version
* [twisted4086.5.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/315/3158e5f03a0a49527162d2089c27901dd431203b/20412eddc2150888da75ff6eb9b79e778ac2177f.patch) (4892 bytes) - added by necaris on 2012-03-13 20:27:00Z - Patch to deprecate "unsigned" flag, now with better docstring

<details><summary>Searchable metadata</summary>

```
trac-id__4086 4086
type__enhancement enhancement
reporter__TimAllen TimAllen
priority__normal normal
milestone__ 
branch__branches_deprecate_tap2rpm_4086 branches/deprecate-tap2rpm-4086
branch_author__cyli cyli
status__closed closed
resolution__fixed fixed
component__core core
keywords__easy easy
time__1258003383000000 1258003383000000
changetime__1331673407000000 1331673407000000
version__None None
owner__ 
cc__thijs cc__necaris
```
</details>

**Comment by twisted-trac** [comment]
|<img alt="TimAllen's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_1"></a>|TimAllen commented|
|-|-|

This ticket split off from [#3292](https://github.com/twisted/twisted/issues/3292).

**Comment by twisted-trac** [comment]
|<img alt="jtanis's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|jtanis set status to `closed`|
|-|-|

Marked for review, sorry about the spam up there.

**Comment by twisted-trac** [comment]
|<img alt="jtanis's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|jtanis set status to `reopened`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="jtanis's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|jtanis set owner to jtanis<br>jtanis set status to `new`|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="jtanis's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|jtanis removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="terrycojones's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|terrycojones commented|
|-|-|

This is going to need a little (easy) work.

 1. The flag shouldn't be removed, instead its use should be deprecated. A ticket can later be opened to remove the flag.
 1. You should add a test/test_tap2rpm.py that imports the MyOptions class and calls parseOptions to make sure the deprecation warning is raised. Take a look in twisted/test/*.py for testing of deprecation warnings. You only need write the single test for the change you're making.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun set owner to jtanis|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="Screwtape's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_9"></a>|Screwtape commented|
|-|-|

Note that [#3292](https://github.com/twisted/twisted/issues/3292) has landed now, and made various changes to the MyOptions class. The "unsigned" command-line option is still there, however, so this ticket still needs fixing.

**Comment by twisted-trac** [comment]
|<img alt="necaris's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|necaris removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_12"></a>|@thijstriemstra set owner to necaris|
|-|-|

Thanks for your patch. Can you include the Twisted version number (12.1) in the deprecation message as you see in other deprecations? It also needs a `.removal` [ReviewProcess#Newsfiles NEWS file].

**Comment by twisted-trac** [comment]
|<img alt="necaris's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_13"></a>|necaris removed owner|
|-|-|

Thanks for the review. I've changed the way the option is handled, so that the twisted.python.deprecate machinery can be used to generate a more standard deprecation warning, and added a NEWS file as requested.

**Comment by twisted-trac** [comment]
|[<img alt="thijstriemstra's avatar" src="https://avatars.githubusercontent.com/u/305679?s=50" width="50" height="50">](https://github.com/thijstriemstra)<a name="note_14"></a>|@thijstriemstra set owner to necaris|
|-|-|

Thanks, the warning references 12.2 instead of 12.1 though.

**Comment by twisted-trac** [comment]
|<img alt="necaris's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_15"></a>|necaris removed owner|
|-|-|

Oops, my mistake -- corrected patch.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_16"></a>|@cyli set owner to necaris|
|-|-|

Thank you for all your hard work necaris!  And thanks so much for replacing all those tabs!

The docstring for opt_XXX in usage.Options subclasses are actually presented to the user whe they ask for usage help.  For example, if you look at the opt_XXXX docstrings in twisted.web.tap.py, and then run twistd web --help, you'll see that they match.

Can you please change the docstring to it's more user-help friendly?  For example, the "Handle the 'unsigned' command line flag" is more like a docstring aimed at developers.

Also, this is a minor nitpick, but there should be two lines between class methods (opt_unsigned has only 1 line between it and the previous method). (for reference, here is the part of the [coding standard](http://twistedmatrix.com/documents/current/core/development/policy/coding-standard.html#auto5) dealing with whitespace).

Once again, thank you so much for working on this!

**Comment by twisted-trac** [comment]
|<img alt="necaris's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_17"></a>|necaris removed owner|
|-|-|

Thank you for your reviews cyli! I've updated the patch, with the right convention regarding spacing and a more helpful docstring (modeled on a deprecated option to twisted.tap.ftp -- hope that was a good one to follow).

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_18"></a>|@cyli commented|
|-|-|

(In [33743]) Branching to 'deprecate-tap2rpm-4086'

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_19"></a>|@cyli commented|
|-|-|

(In [33744]) Apply patch from necaris (with newsfile from previous patch)

refs [#4086](https://github.com/twisted/twisted/issues/4086)

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_20"></a>|@cyli commented|
|-|-|

Thanks for your prompt responses necaris!  This looks great ([buildbot results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/deprecate-tap2rpm-4086)).

Going to merge into trunk.

**Comment by twisted-trac** [comment]
|[<img alt="cyli's avatar" src="https://avatars.githubusercontent.com/u/223200?s=50" width="50" height="50">](https://github.com/cyli)<a name="note_21"></a>|@cyli set status to `closed`|
|-|-|

(In [33747]) Merge deprecate-tap2rpm-4086: deprecate the --unsigned option in tap2rpm

Author: necaris
Reviewer: thijs, cyli
Fixes: [#4086](https://github.com/twisted/twisted/issues/4086)

The --unsigned option in tap2rpm is now deprecated

**Comment by twisted-trac** [comment]
|<img alt="necaris's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_22"></a>|necaris commented|
|-|-|

Thank you guys for your patient reviews while I fumbled my way towards a usable patch!
